wallet/core/src/message.rs

```diff
@@ -0,0 +1,265 @@
+use kaspa_hashes::{Hash, PersonalMessageSigningHash};
+use secp256k1::{Error, XOnlyPublicKey};
+
+#[derive(Clone)]
+pub struct PersonalMessage<'a>(&'a str);
+
+impl AsRef<[u8]> for PersonalMessage<'_> {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+pub fn sign_message(msg: PersonalMessage, privkey: &[u8; 32]) -> Result<Vec<u8>, Error> {
+    let hash = calc_personal_message_hash(msg);
+
+    let msg = secp256k1::Message::from_slice(hash.as_bytes().as_slice())?;
+    let schnorr_key = secp256k1::KeyPair::from_seckey_slice(secp256k1::SECP256K1, privkey)?;
+    let sig: [u8; 64] = *schnorr_key.sign_schnorr(msg).as_ref();
+
+    Ok(sig.to_vec())
+}
+
+pub fn sign_message_with_aux_rand(msg: PersonalMessage, privkey: &[u8; 32], aux_rand: &[u8; 32]) -> Result<Vec<u8>, Error> {
+    let hash = calc_personal_message_hash(msg);
+
+    let msg = secp256k1::Message::from_slice(hash.as_bytes().as_slice())?;
+    let schnorr_key = secp256k1::KeyPair::from_seckey_slice(secp256k1::SECP256K1, privkey)?;
+    let curve = secp256k1::Secp256k1::new();
+    let sig: [u8; 64] = *curve.sign_schnorr_with_aux_rand(&msg, &schnorr_key, aux_rand).as_ref();
+
+    Ok(sig.to_vec())
+}
+
+pub fn verify_message(msg: PersonalMessage, signature: Vec<u8>, pubkey: XOnlyPublicKey) -> Result<bool, Error> {
+    let hash = calc_personal_message_hash(msg);
+    let msg = secp256k1::Message::from_slice(hash.as_bytes().as_slice())?;
+    let sig = secp256k1::schnorr::Signature::from_slice(signature.as_slice())?;
+    let sig_res = sig.verify(&msg, &pubkey);
+
+    match sig_res {
+        Ok(_) => Ok(true),
+        Err(e) => Err(e),
+    }
+}
+
+fn calc_personal_message_hash(msg: PersonalMessage) -> Hash {
+    let mut hasher = PersonalMessageSigningHash::new();
+    hasher.write(msg);
+    hasher.finalize()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_basic_sign_and_verify_sign() {
+        let pm = PersonalMessage("Hello Kaspa!");
+        let privkey: [u8; 32] = [
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03,
+        ];
+        let pubkey = XOnlyPublicKey::from_slice(&[
+            0xF9, 0x30, 0x8A, 0x01, 0x92, 0x58, 0xC3, 0x10, 0x49, 0x34, 0x4F, 0x85, 0xF8, 0x9D, 0x52, 0x29, 0xB5, 0x31, 0xC8, 0x45,
+            0x83, 0x6F, 0x99, 0xB0, 0x86, 0x01, 0xF1, 0x13, 0xBC, 0xE0, 0x36, 0xF9,
+        ])
+        .unwrap();
+        let sig_result = sign_message(pm.clone(), &privkey);
+
+        assert!(sig_result.is_ok());
+
+        if let Ok(sig_result) = sig_result {
```

Comment:
opinions might differ, but although I myself do `if let Ok()`, pre-checking `assert!()` and then doing `if let Ok()` is a bit redundant. Following are few suggestions on alternate handling:

since `assert!()` triggers a `panic!()` (which is what `expect()` does as well, may I suggest something like:
```
let sig = sig_result.expect("signature result error");
verify_ message(pm,sig_result,pubkey).expect("signature verification error");
```
this can be further collapsed to:
```
verify_message(
    pm.clone(), 
    sign_message(pm, &privkey).expect("sign_message failure"), 
    pubkey)
.expect("verify_message failure");
```
this is more rusty.

I think `expect()` would be better here as it signals what is actually wrong.